### PR TITLE
Updated AJAX request to avoid old style coding

### PR DESCRIPTION
**Pull request by FiwareULPGC**
ActiveX Ajax code has been deleted. 
Now '.onchange' is used instead of 'onreadystatechange' while performing AJAX requests.

This Pull Request should solve #17 .

### CODE REVIEW

File: client/js/js_utils.js [hunk 1]
```diff
@@ -24,198 +24,175 @@ function addLoadEvent(func) {
...
+  {
+    if (200 == this.status)
+    {
+      on_success_callback(this.response);
+    }
+    else
+    {
+      try
+      {
+        on_failure_callback();
```
Comment (jmcanterafonseca) [inline client/js/js_utils.js]: ooopssss 

why do you need this? that's not Javascript proper style

File: client/js/js_utils.js [hunk 2]
```diff
@@ -24,198 +24,175 @@ function addLoadEvent(func) {
...
+      }
+      catch (err)
+      {
+        on_failure_callback(this.response);
+      }
```
Comment (jmcanterafonseca) [inline client/js/js_utils.js]: please avoid that

**Comment by FiwareULPGC** [comment]
Unnecessary try-catch blocks have been removed.

File: client/js/js_utils.js [hunk 3]
```diff
@@ -26,181 +26,97 @@ function addLoadEvent(func) {
...
-  xmlhttp.setRequestHeader('Fiware-Service', 'tourguide');
-  xmlhttp.setRequestHeader('Content-type', 'application/json');
-  xmlhttp.send(JSON.stringify(data));
-}
 
+  xhr.onerror = function(e) {
+      on_failure_callback(this.response);
+  };
+  xhr.open('PATCH', url, true);
+  xhr.setRequestHeader('Fiware-Service', 'tourguide');
```
Comment (jmcanterafonseca) [inline client/js/js_utils.js]: this code is repeated all over many places.

Why don't you create an xhr builder a centralize this process in one place

File: client/js/js_utils.js [hunk 4]
```diff
@@ -26,181 +26,97 @@ function addLoadEvent(func) {
...
-          on_failure_callback(xmlhttp.responseText);
-        }
-      }
+  xhr.onload = function(e) {
+    if (200 == this.status) {
+      on_success_callback(this.response);
+    }
+    else {
+      on_failure_callback(this.response);
     }
```
Comment (jmcanterafonseca) [inline client/js/js_utils.js:84]: does it make sense to pass the response when there has been failure? what is the point?

Comment (FiwareULPGC) [inline client/js/js_utils.js:84]: The reason is that failures also have responses. We usually use it to process errors or show debugs messages.

For instance, if we perform a reservation like

```
post_ajax_petition(base_url + 'reservation/',
    closePopUpWindow,
    function(err) {
      alert('Cannot add reservation');
      console.log(err);
    }, data);
```

without the fix #28, it fails and we get the console message:

```
restaurants_api.js:687:7 
{"error":{"message":"Invalid type: string (expected number)","code":400,"params":{"type":"string","expected":"number"},"dataPath":"/partySize","title":"Bad request"}} 
```

File: client/js/js_utils.js (hunk 3 above)
Comment (FiwareULPGC) [inline client/js/js_utils.js]: Thank you, it is really a good idea! We are going to integrate it into the next iteration of the PR.

Comment (jmcanterafonseca) [inline client/js/js_utils.js]: I think we can do it in this PR

File: client/js/js_utils.js [hunk 5]
```diff
@@ -26,181 +26,97 @@ function addLoadEvent(func) {
...
+  xhr.onerror = function(e) {
+      on_failure_callback(this.response);
+  };
+  xhr.open('PATCH', url, true);
+  xhr.setRequestHeader('Fiware-Service', 'tourguide');
+  xhr.setRequestHeader('Content-type', 'application/json');
+  xhr.send(JSON.stringify(data));
+}
 
 Date.prototype.yyyymmdd = function() {
```
Comment (jmcanterafonseca) [inline client/js/js_utils.js]: not a good idea to extend built-in objects

File: client/js/js_utils.js (hunk 4 above)
Comment (jmcanterafonseca) [inline client/js/js_utils.js:84]: ok, thanks

File: client/js/js_utils.js (hunk 5 above)
Comment (FiwareULPGC) [inline client/js/js_utils.js]: This “utility” is planned to be deleted in a future PR (it is only used by one function). We suggest to keep this PR  with only changes related to AJAX. If we delete it in this PR, several conflicts will appear and should be solved.

**Comment by FiwareULPGC** [comment]
PR updated. Duplicity of code is avoided. yyymmdd method for date objects was deleted and it is no longer used.
If you agree with the PR we can rebase it.

**Comment by jmcanterafonseca** [comment]
LGTM

**Pull request merged**